pdns/minicurl.cc

```diff
@@ -104,7 +118,7 @@ static string extractHostFromURL(const std::string& url)
 
 void MiniCurl::setupURL(const std::string& str, const ComboAddress* rem, const ComboAddress* src, int timeout, size_t byteslimit, bool fastopen, bool verify)
 {
```

Comment:
I *think* we should `clearHostsList()` first

Reply:
That's a nice one! It looks like we are good if the call is passing a non-null pointer for `rem`, as we would then completely replace the list, starting with an empty one, but we would indeed keep the list around if `rem` which doesn't seem good. I'll fix it, thanks!

Reply:
Digging a bit more, I don't think we can currently reuse a `MiniCurl` object for more requests, as we are not calling `curl_easy_reset`. Technically it should be fine as long as we use the same parameters, but we might be leaking memory.
Are you aware of any place where we reuse these objects?

Reply:
I'm going to push an attempt at making it safe to reuse a `MiniCurl` object, as this is clearly something that might happen